sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs

```diff
@@ -152,8 +152,7 @@ void AddNonNullPolicies(HttpPipelinePolicy[] policiesToAdd)
                 policies.Add(CreateTelemetryPolicy(buildOptions.ClientOptions));
             }
 
-            RetryOptions retryOptions = buildOptions.ClientOptions.Retry;
-            policies.Add(new RetryPolicy(retryOptions.Mode, retryOptions.Delay, retryOptions.MaxDelay, retryOptions.MaxRetries));
+            policies.Add(buildOptions.ClientOptions.RetryPolicy ?? new DefaultRetryPolicy(buildOptions.ClientOptions.Retry));
```

Comment:
Does this honor the static client options?

Reply:
Good catch.

Reply:
Fixed, will need to add tests.